payjoin-relay/src/db.rs

```diff
@@ -1,189 +1,355 @@
-use std::time::Duration;
+#![allow(unused_imports)]
+use std::{
+    fmt::Display,
+    pin::Pin,
+    sync::{Arc, Mutex},
+    thread,
+    time::Duration,
+};
 
 use anyhow::Result;
-use sqlx::postgres::{PgListener, PgPoolOptions};
-use sqlx::{PgPool, Pool, Postgres};
-use tracing::debug;
+use futures_util::StreamExt;
+use redis::{
+    aio::{self, AsyncStream},
+    AsyncCommands, Client, Commands, Connection, ControlFlow, PubSub, PubSubCommands, RedisError,
+};
 
-const RES_COLUMN: &str = "res";
-const REQ_COLUMN: &str = "req";
+pub type DatabaseClient = Arc<Mutex<Client>>;
 
-pub(crate) struct DbPool {
-    pool: Pool<Postgres>,
-    timeout: Duration,
+pub trait RedisPool {
+    fn client(&self) -> &DatabaseClient;
+    // fn sub(&self);
+    fn read(&self, channel: &str, pubkey_id: &str) -> Result<Vec<u8>, RedisError>;
```

Comment:
Same with exposing RedisErrors. The upstream app shouldn't even know it's dealing with redis. Just that it wants to push and peek data.

Reply:
I think this comment was misguided. I see now that existing methods exposed sqlx::Errors, and this can be handled upstream without actually understanding the redis details by e.g. just printing the error.